### PR TITLE
Fix ParallelMLP and enable accelerator test

### PR DESCRIPTION
**Pull request by xinyu-intel**
- minor fix in ParallelMLP

- apply https://github.com/microsoft/Megatron-DeepSpeed/pull/104 to more test cases

### CODE REVIEW

**Comment by xinyu-intel** [comment]
cc @polisettyvarma

File: tests/transformer/test_parallel_mlp.py [hunk 1]
```diff
@@ -19,28 +21,27 @@ def test_constructor(self, mlp):
...
-        hidden_states = hidden_states.cuda()
+        hidden_states = torch.ones((32, 2, transformer_config.hidden_size))
+        hidden_states = hidden_states.to(device_name)
         output, output_bias = mlp(hidden_states)
         assert output.shape[0] == 32
         assert output.shape[1] == 2
-        assert output.shape[2] == mlp.config.hidden_size
-        assert output_bias.shape[0] == mlp.config.hidden_size
+        assert output.shape[2] == transformer_config.hidden_size
+        assert output_bias == None
```
Comment (tjruwase) [inline tests/transformer/test_parallel_mlp.py:44]: Why is this change needed?

File: tests/transformer/test_parallel_mlp.py [hunk 2]
```diff
@@ -19,28 +21,27 @@ def test_constructor(self, mlp):
         num_weights = sum([p.numel() for p in mlp.parameters()])
         assert num_weights == 1212
 
-    def test_cpu_forward(self, mlp):
+    def test_cpu_forward(self, mlp, transformer_config):
         # [sequence length, micro batch size, hidden size]
-        hidden_states = torch.ones((32, 2, mlp.config.hidden_size))
+        hidden_states = torch.ones((32, 2, transformer_config.hidden_size))
```
Comment (tjruwase) [inline tests/transformer/test_parallel_mlp.py:26]: Please explain why `transformer_config` should be added here?

Comment (xinyu-intel) [reply tests/transformer/test_parallel_mlp.py:26]: mlp, a.k.a constructed model, doesn't have config as member. So, here transformer_config is to get the proper hidden_size.

File: tests/transformer/test_parallel_mlp.py (hunk 1 above)
Comment (tjruwase) [reply tests/transformer/test_parallel_mlp.py:44]: Thanks for the explanation. But should this be conditional based on
https://github.com/microsoft/Megatron-DeepSpeed/blob/7eb36a11b3a9c48ed07b93692ccf22bfb5577f7e/megatron/core/tensor_parallel/layers.py#L484

File: tests/transformer/test_parallel_mlp.py (hunk 2 above)
Comment (tjruwase) [reply tests/transformer/test_parallel_mlp.py:26]: That makes sense. Is `tranformer_config` a pytest fixture?

Comment (xinyu-intel) [reply tests/transformer/test_parallel_mlp.py:26]: yes. https://github.com/microsoft/Megatron-DeepSpeed/blob/main/tests/conftest.py#L21

File: tests/transformer/test_parallel_mlp.py (hunk 1 above)
Comment (xinyu-intel) [reply tests/transformer/test_parallel_mlp.py:44]: right. but hard to get the value from the test because it's hardcoded in the https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/model/transformer.py#L117. Any ideas?

Comment (tjruwase) [reply tests/transformer/test_parallel_mlp.py:44]: Got it. Thanks for the explanation. I don't see any easy way to resolve this, and I don't think improving this test coverage is worth the trouble.

**Review by tjruwase — APPROVED** [review]
